Dev2

angeldanielboscan opened the pull request.

angeldanielboscan committed:
Update supervisord-nginx.conf

angeldanielboscan committed:
Improving cron support [NOT FULLY TESTED]

angeldanielboscan committed:
Bug fix of cron improvement, adding CRON_SCHEDULE variable for setting cron jobs

angeldanielboscan committed:
Adding .htaccess files according https://www.drupal.org/node/2475373

angeldanielboscan commented on startup.sh:
```diff
@@ -56,3 +56,62 @@ if [[ -z $CHECK ]]; then
+
+# Adding .htaccess to sites/default/files/ and /tmp according https://www.drupal.org/SA-CORE-2013-003
+
+FILES_HTACCESS="# Turn off all options we don't need.
```
Tal vez sea mejor colocar estas validaciones en otro archivo aparte.

c3s4r commented on startup.sh:
```diff
@@ -36,4 +36,82 @@ echo "ff02::1	ip6-allnodes" >> /etc/hosts
 # Runnning supervisor
-/usr/bin/supervisord -n
\ No newline at end of file
+/usr/bin/supervisord -n
```
Creo que esta debe ser la úlltima línea, porque estamos ejectuando supervisord en foreground, por tanto las siguientes líneas no se van a ejecutar.

angeldanielboscan replied:
Las líneas de código posteriores a a la ejecución de supervisord corren sin problemas (extraño!), igualmente lo colocaré de la forma que comentas ya que es la correcta.

c3s4r commented on startup.sh:
```diff
@@ -36,4 +36,82 @@ echo "ff02::1	ip6-allnodes" >> /etc/hosts
+CHECK=$(cat /etc/crontab | grep -o "$CRON_JOB" )
+
+if [[ -z $CHECK ]]; then
+	echo "$CRON_SCHEDULE www-data /usr/bin/env PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin COLUMNS=72 /usr/local/bin/drush --root=/var/www cron" >> /etc/crontab
```
Se podría reemplazar por esto?

`echo "$CRON_SCHEDULE $CRON_JOB" >> /etc/crontab`

c3s4r commented on startup.sh:
```diff
@@ -36,4 +36,82 @@ echo "ff02::1	ip6-allnodes" >> /etc/hosts
+	if [[ "$FILES_HTACCESS" = $(cat htaccessfiles.txt) ]]; then
+		echo "File already exists"
+	else
+		echo "Files different"
```
Solo deberíamos crear los archivos si es que no existen. Si es que ya existen, no deberíamos modificarlos.

angeldanielboscan committed:
Working drush cron job

Adding env variables to /etc/cron.d/drupal, this allows Drupal Drush cron to handle the ENV variables in the container

angeldanielboscan committed:
Changing startup.sh to write .htaccess files only if they don't exists

angeldanielboscan committed:
Moving supervisor command to the bottom of startup.sh

c3s4r commented on startup.sh:
```diff
@@ -35,5 +41,73 @@ echo "ff00::0	ip6-mcastprefix" >> /etc/hosts
+  php_flag engine off
+</IfModule>"
+
+TMP_HTACCESS="# Turn off all options we don't need.
```
No estoy seguro, pero me parece que el htaccess de tmp era mucho más sencillo. Estas seguro que estamos usando el código correcto?

angeldanielboscan replied:
Según 
https://www.drupal.org/node/2140629 y https://www.drupal.org/SA-CORE-2013-003

Al parecer, archivo en /tmp/.htaccess es igual al de files/ pero añadiendo el comando "Deny from all"

angeldanielboscan committed:
Update README.md
